applications/plugins/SofaCarving/examples/SimpleCarving.py

```diff
@@ -0,0 +1,100 @@
+   
+    TT.addObject('MechanicalObject',template="Vec3d",src="@loader",name="Volume")
+    
+    TT.addObject('TetrahedronSetTopologyContainer', name="topo", src="@loader")
+    TT.addObject('TetrahedronSetTopologyModifier', name="topoMod")
+    TT.addObject('TetrahedronSetGeometryAlgorithms', template="Vec3d", name="GeomAlgo")
+    
+    TT.addObject('DiagonalMass',massDensity="0.5")
+    TT.addObject('FixedConstraint',indices="1 3 50")
+    TT.addObject('TetrahedralCorotationalFEMForceField',name="CFEM",youngModulus="160",poissonRatio="0.3",method="large")
+    
+    T=TT.addChild('T')    
+    T.addObject('TriangleSetTopologyContainer', name="Container")
+    T.addObject('TriangleSetTopologyModifier', name="Modifier")
+    T.addObject('TriangleSetGeometryAlgorithms',name="GeomAlgo",template="Vec3d")
+    T.addObject('Tetra2TriangleTopologicalMapping', input="@../topo", output="@Container")
+    
+    T.addObject('TriangleCollisionModel', tags="CarvingSurface")
+
+    Visu = T.addChild('Visu')
+    Visu.addObject('OglModel',name="Visual",material="Default Diffuse 1 0 1 0 1 Ambient 0 1 1 1 1 Specular 1 1 1 0 1 Emissive 0 1 1 0 1 Shininess 1 100")
+    Visu.addObject('IdentityMapping',input="@Volume",output="@Visual")
+
+
+    Instrument = root.addChild('Instrument')
+    Instrument.addObject('EulerImplicitSolver',name="cg_odesolver",printLog="false")
+    Instrument.addObject('CGLinearSolver',iterations="25",name="linear solver",tolerance="1.0e-9",threshold="1.0e-9")
+    Instrument.addObject('MechanicalObject',template="Rigid3d",name="instrumentState",tags="Omni",rotation="90 45 0",translation="0 0 1")
+    Instrument.addObject('UniformMass',template="Rigid3d",name="mass",totalMass="5")
+    visuIns = Instrument.addChild('visuIns')
+    visuIns.addObject('MeshOBJLoader',name="meshLoader_0",filename="mesh/dental_instrument_light.obj",scale3d="1 1 1",translation="-0.412256 -0.067639 3.35",rotation="180 0 150",handleSeams="1")
+    visuIns.addObject('OglModel',template="Vec3d",name="InstrumentVisualModel",src="@meshLoader_0",material="Default Diffuse 1 1 0.2 0.2 1 Ambient 1 0.2 0.04 0.04 1 Specular 0 1 0.2 0.2 1 Emissive 0 1 0.2 0.2 1 Shininess 0 45")
+    visuIns.addObject('RigidMapping',template="Rigid3d,Vec3d",name="MM->VM mapping",input="@instrumentState",output="@InstrumentVisualModel")
+
+    colIns = Instrument.addChild('colIns')
+    colIns.addObject('MechanicalObject',template="Vec3d",name="Particle",position="-0.2 -0.2 -0.2")
+    colIns.addObject('SphereCollisionModel',name="ParticleModel",radius="0.2",tags="CarvingTool")
+    #colIns.addObject('SphereCollisionModel',name="ParticleModel",radius="0.2")
+    colIns.addObject('RigidMapping',template="Rigid3d,Vec3d",name="MM->CM mapping",input="@instrumentState",output="@Particle")
+
+    return root
+
+
+#Function used only if this script is called from a python environment
+if __name__ == '__main__':
+    main()
```

Comment:
Instead of a global variable, please do something like 
```python
if __name__ == '__main__':
    with_gui = False
    if len(sys.args) == 2:
        with_gui = sys.args[1] == "with-gui"
    main(with_gui)
```
Would be even better to use argparser https://docs.python.org/3/library/argparse.html instead of manually processing the args.